core/trino-main/src/main/java/io/trino/sql/planner/LogicalPlanner.java

```diff
@@ -605,6 +608,12 @@ private RelationPlan createTableWriterPlan(
     {
         Optional<PartitioningScheme> partitioningScheme = Optional.empty();
         Optional<PartitioningScheme> preferredPartitioningScheme = Optional.empty();
+
+        int maxWriterTasks = target.getMaxWriterTasks(plannerContext.getMetadata(), session).orElse(getMaxWriterTasksCount(session));
+        Optional<Integer> maxWritersNodesCount = getRetryPolicy(session) != RetryPolicy.TASK
```

Comment:
assert that when `getRetryPolicy(session) == RetryPolicy.TASK` then `getMaxWriterTasks` is empty.
Alternatively add comment to `getMaxWriterTasks` that it's ignored when for `RetryPolicy.TASK`